src/System.ServiceModel.Federation/WsTrustChannelSecurityTokenManager.cs

```diff
@@ -26,16 +36,26 @@ public override SecurityTokenProvider CreateSecurityTokenProvider(SecurityTokenR
         {
             // If token requirement matches SAML token return the custom SAML token provider
             // that performs custom work to serve up the token
-            var tokenProvider = (ClientCredentials is WsTrustChannelClientCredentials wsTrustChannelClientCredentials) ?
-                new WSTrustChannelSecurityTokenProvider(tokenRequirement, wsTrustChannelClientCredentials.RequestContext)
+            if (tokenRequirement.TokenType.Equals(Saml2Constants.OasisWssSaml2TokenProfile11))
```

Comment:
This should work for both SAML 2 and 1.1. Should we check whether the TokenType is `SamlConstants.OasisWssSamlTokenProfile11` too?

Reply:
there are different values that indicated see: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/34a097ed2a97f1c8d06722bc6d6dca0594f0ed62/src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Constants.cs#L41

Reply:
I'll add all three Saml and Saml2 token types.